### PR TITLE
Building relationships

### PR DESCRIPTION
**Pull request by jardrake**

### CODE REVIEW

**Review by jardrake — COMMENTED** [review]
Many thanks. This is a much better way to do it

File: src/API/Data/BuildingRelationshipsRepository.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+			}
+			//Unit is already being checked/found in the Authorization step
+			if (await db.Buildings.AnyAsync(b => b.Id == body.BuildingId) == false)
+			{
+				return Pipeline.NotFound("The specified unit and/or building does not exist.");
+			}
+			if (await db.BuildingRelationships.AnyAsync(r => r.BuildingId == body.BuildingId && r.UnitId == body.UnitId))
+			{
+				return Pipeline.Conflict("The provided unit already has a support relationship with the provided building.");
+			}
```
Comment (jhoerr) [inline src/API/Data/BuildingRelationshipsRepository.cs:93]: What happens here if I create a relationship, and then update it with the same properties?

**Review by jhoerr — COMMENTED** [review]
Looks awesome! I have one minor question about the request validation.

**Review by jasonfrancis — APPROVED** [review]

File: src/API/Data/BuildingRelationshipsRepository.cs (hunk 1 above)
Comment (jardrake) [reply src/API/Data/BuildingRelationshipsRepository.cs:93]: That's a good point. I'll fix this to check if the existing match is the same one that's being updated. If it is, we can just return an ok.

Comment (jhoerr) [reply src/API/Data/BuildingRelationshipsRepository.cs:93]: Works for me. Thanks!

**Review by jhoerr — APPROVED** [review]

**Pull request merged**